package.json

```diff
@@ -10,19 +10,32 @@
     "license": "ISC",
     "homepage": "https://github.com/quentinlao/reactTemplate#readme",
     "devDependencies": {
-        "@babel/core": "^7.16.12",
+        "@babel/core": "^7.17.2",
         "@babel/plugin-proposal-class-properties": "^7.16.7",
+        "@babel/preset-env": "^7.16.11",
         "@babel/preset-react": "^7.16.7",
+        "@babel/preset-typescript": "^7.16.7",
+        "@cypress/code-coverage": "^3.9.12",
+        "@cypress/webpack-preprocessor": "^5.11.1",
+        "@testing-library/jest-dom": "^5.16.2",
+        "@testing-library/react": "^12.1.2",
+        "@types/chai": "^4.3.0",
+        "@types/jest": "^27.4.0",
         "@typescript-eslint/eslint-plugin": "^5.10.2",
         "@typescript-eslint/parser": "^5.10.2",
+        "babel-jest": "^27.5.1",
         "babel-loader": "^8.2.3",
         "babel-preset-react": "^6.24.1",
+        "chai": "^4.3.6",
         "css-loader": "^6.5.1",
+        "cypress": "9.5.0",
         "dotenv": "^15.0.0",
         "eslint": "^8.8.0",
         "eslint-plugin-react": "^7.28.0",
         "eslint-plugin-react-hooks": "^4.3.0",
         "html-webpack-plugin": "^5.5.0",
+        "jest": "^27.5.1",
+        "nyc": "^13.3.0",
         "prettier": "2.5.1",
         "style-loader": "^3.3.1",
         "ts-loader": "^9.2.6",
```

Comment:
I think the dependence is not used anymore.
babel-loader has taken its place

Reply:
right ts loader is unused